src/main/java/com/rebuy/consul/listeners/EmbeddedServletContainerInitializedEventListener.java

```diff
@@ -0,0 +1,21 @@
+package com.rebuy.consul.listeners;
+
+import com.rebuy.consul.ConsulService;
+import org.springframework.boot.context.embedded.EmbeddedServletContainerInitializedEvent;
+import org.springframework.context.ApplicationListener;
+
+public class EmbeddedServletContainerInitializedEventListener implements ApplicationListener<EmbeddedServletContainerInitializedEvent>
```

Comment:
even though the docu is old:
http://docs.spring.io/autorepo/docs/spring-boot/1.2.0.M2/api/org/springframework/boot/context/embedded/EmbeddedServletContainerInitializedEvent.html
it is called after refresh ready so it should be better.